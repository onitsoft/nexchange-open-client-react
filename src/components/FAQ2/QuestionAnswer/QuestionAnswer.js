import React, { Component } from 'react';
import { translate } from 'react-i18next';

import styles from './QuestionAnswer.scss';

class QuestionAnswer extends Component {
  constructor(props) {
    super(props);

    this.state = { open: false, positiveFeedback: false, negativeFeedback: false };

    this.setPositiveFeedback = this.setPositiveFeedback.bind(this);
    this.setNegativeFeedback = this.setNegativeFeedback.bind(this);
  }

  componentDidMount() {
    const browserPath = window.location.pathname.split('/')[2]
      ? window.location.pathname
          .split('/')[2]
          .toLowerCase()
          .trim()
      : null;
    /* eslint max-len: ["error", { "code": 200 }] */

    const questionPath = this.props
      .t(`faq.${this.props.id}`)
      .replace(/ /g, '-')
      .replace(/[&/\\#,+()$~%.'":*?<>{}]/g, '')
      .toLowerCase()
      .trim();
    if (browserPath === questionPath || decodeURIComponent(browserPath).toLowerCase() === questionPath) {
      this.setState({ open: true });
      this.shouldScrollToElement = true;
    }
  }

  componentDidUpdate() {
    if (this.shouldScrollToElement && !this.scrolledToElement) {
      this.scrolledToElement = true;
      document.getElementById(`faq.${this.props.id}`).scrollIntoView();
    }
  }

  onClick = () => {
    this.setState({ open: !this.state.open });
    window.gtag('event', 'Question open', { event_category: 'FAQ', event_label: `${this.props.t(`faq.${this.props.id}`)}` });
  };

  setPositiveFeedback = () => {
    if (!this.state.positiveFeedback) {
      window.gtag('event', 'Positive Feedback', { event_category: 'FAQ', event_label: `${this.props.t(`faq.${this.props.id}`)}` });
      this.setState({ positiveFeedback: true, negativeFeedback: false });
    }
  };

  setNegativeFeedback = () => {
    if (!this.state.negativeFeedback) {
      window.gtag('event', 'Negative Feedback', { event_category: 'FAQ', event_label: `${this.props.t(`faq.${this.props.id}`)}` });
      this.setState({ positiveFeedback: false, negativeFeedback: true });
      this.props.openSupportModal(this.props.question || this.props.t(`faq.${this.props.id}`));
    }
  };

  render() {
    return (
      <div className={`question-answer ${styles.container}`} id={`faq.${this.props.id}`}>
        <div className={`${styles.question}`} onClick={this.onClick} data-test="question-opener">
          <i className={`far fa-${this.state.open ? 'minus' : 'plus'}-square fa-2x`} aria-hidden="true" />
          <h3>{this.props.question || this.props.t(`faq.${this.props.id}`)}</h3>
        </div>
        <div className={`${this.state.open ? `${styles.answer} ${styles.active}` : `${styles.answer}`}`}>
          {this.props.answer}
          <div className={styles.feedback}>
            <div>
              <span>Was this helpful?</span>
            </div>
            <div className={styles.buttons}>
              <button
                className={`btn ${styles.positive} ${this.state.positiveFeedback ? styles.active : ''}`}
                onClick={() => {
                  this.setPositiveFeedback();
                }}
              >
                <i className="fas fa-thumbs-up"></i>
                <span>YES</span>
              </button>
              <button
                className={`btn ${styles.negative} ${this.state.negativeFeedback ? styles.active : ''}`}
                onClick={() => {
                  this.setNegativeFeedback();
                }}
              >
                <i className="fas fa-thumbs-down"></i>
                <span>NO</span>
              </button>
            </div>
          </div>
        </div>
        <hr />
      </div>
    );
  }
}

export default translate()(QuestionAnswer);
