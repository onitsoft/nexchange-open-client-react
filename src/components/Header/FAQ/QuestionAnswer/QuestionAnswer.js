import React, { Component } from 'react';
import { translate } from 'react-i18next';

class QuestionAnswer extends Component {
  state = { open: false };

  onClick = () => {
    this.setState({ open: !this.state.open });
    window.gtag('event', 'question open', { 'event_category' : 'FAQ' });
  };

  render() {
    return (
      <div className="question-answer">
        <h3 onClick={this.onClick} data-test="question-opener">
          {this.props.t(`faq.${this.props.id}`)} <i className={`far fa-${this.state.open ? 'minus' : 'plus'}-square`} aria-hidden="true" />
        </h3>
        <div className={this.state.open ? 'answer active' : 'answer'}>{this.props.answer}</div>
        <hr />
      </div>
    );
  }
}

export default translate()(QuestionAnswer);
