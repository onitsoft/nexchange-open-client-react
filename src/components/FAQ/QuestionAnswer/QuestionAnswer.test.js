import React from 'react';
import { shallow } from 'enzyme';
import QuestionAnswer from './QuestionAnswer';

describe('QuestionAnswer', () => {
  let wrapShallow;

  beforeEach(() => {
    window.ga = jest.fn();

    wrapShallow = shallow(
      <QuestionAnswer
      id="ques1"
      answer={
          <div>
            <p>
              We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin
              selection widget in the top of fold of the page.
            </p>
            <p>We will be adding more currencies very soon, stay tuned!</p>
            <p>
              Missing your favorite coin? Let us know here: <a href="mailto:support@nexchange.co.uk">support@nexchange.co.uk</a>.
            </p>
          </div>
        }
      />
    ).dive();
  });

  it('renders correctly', () => {
    expect(wrapShallow).toMatchSnapshot();
  });

  it('question opens on click', () => {
    wrapShallow.find('[data-test="question-opener"]').simulate('click');
    expect(wrapShallow.state().open).toBe(true);
  });
});
