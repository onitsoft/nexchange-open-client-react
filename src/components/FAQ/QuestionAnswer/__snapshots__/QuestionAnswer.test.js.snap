// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionAnswer renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionAnswer
    answer={
      <div>
        <p>
          We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
        </p>
        <p>
          We will be adding more currencies very soon, stay tuned!
        </p>
        <p>
          Missing your favorite coin? Let us know here: 
          <a
            href="mailto:support@nexchange.co.uk"
          >
            support@nexchange.co.uk
          </a>
          .
        </p>
      </div>
    }
    id="ques1"
    t={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="question"
          data-test="question-opener"
          onClick={[Function]}
        >
          <i
            aria-hidden="true"
            className="far fa-plus-square fa-2x"
          />
          <h3>
            faq.ques1
          </h3>
        </div>,
        <div
          className="answer"
        >
          <div>
            <p>
              We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
            </p>
            <p>
              We will be adding more currencies very soon, stay tuned!
            </p>
            <p>
              Missing your favorite coin? Let us know here: 
              <a
                href="mailto:support@nexchange.co.uk"
              >
                support@nexchange.co.uk
              </a>
              .
            </p>
          </div>
          <div
            className="feedback"
          >
            <div>
              <span>
                Whas this helpful?
              </span>
            </div>
            <div
              className="buttons"
            >
              <button
                className="btn positive "
                onClick={[Function]}
              >
                <i
                  className="fas fa-thumbs-up"
                />
                <span>
                  YES
                </span>
              </button>
              <button
                className="btn negative "
                onClick={[Function]}
              >
                <i
                  className="fas fa-thumbs-down"
                />
                <span>
                  NO
                </span>
              </button>
            </div>
          </div>
        </div>,
        <hr />,
      ],
      "className": "question-answer container",
      "id": "faq.ques1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              aria-hidden="true"
              className="far fa-plus-square fa-2x"
            />,
            <h3>
              faq.ques1
            </h3>,
          ],
          "className": "question",
          "data-test": "question-opener",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "far fa-plus-square fa-2x",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "faq.ques1",
            },
            "ref": null,
            "rendered": "faq.ques1",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <p>
                We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
              </p>
              <p>
                We will be adding more currencies very soon, stay tuned!
              </p>
              <p>
                Missing your favorite coin? Let us know here: 
                <a
                  href="mailto:support@nexchange.co.uk"
                >
                  support@nexchange.co.uk
                </a>
                .
              </p>
            </div>,
            <div
              className="feedback"
            >
              <div>
                <span>
                  Whas this helpful?
                </span>
              </div>
              <div
                className="buttons"
              >
                <button
                  className="btn positive "
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-thumbs-up"
                  />
                  <span>
                    YES
                  </span>
                </button>
                <button
                  className="btn negative "
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-thumbs-down"
                  />
                  <span>
                    NO
                  </span>
                </button>
              </div>
            </div>,
          ],
          "className": "answer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
                </p>,
                <p>
                  We will be adding more currencies very soon, stay tuned!
                </p>,
                <p>
                  Missing your favorite coin? Let us know here: 
                  <a
                    href="mailto:support@nexchange.co.uk"
                  >
                    support@nexchange.co.uk
                  </a>
                  .
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.",
                },
                "ref": null,
                "rendered": "We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "We will be adding more currencies very soon, stay tuned!",
                },
                "ref": null,
                "rendered": "We will be adding more currencies very soon, stay tuned!",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Missing your favorite coin? Let us know here: ",
                    <a
                      href="mailto:support@nexchange.co.uk"
                    >
                      support@nexchange.co.uk
                    </a>,
                    ".",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Missing your favorite coin? Let us know here: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "support@nexchange.co.uk",
                      "href": "mailto:support@nexchange.co.uk",
                    },
                    "ref": null,
                    "rendered": "support@nexchange.co.uk",
                    "type": "a",
                  },
                  ".",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <span>
                    Whas this helpful?
                  </span>
                </div>,
                <div
                  className="buttons"
                >
                  <button
                    className="btn positive "
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-thumbs-up"
                    />
                    <span>
                      YES
                    </span>
                  </button>
                  <button
                    className="btn negative "
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-thumbs-down"
                    />
                    <span>
                      NO
                    </span>
                  </button>
                </div>,
              ],
              "className": "feedback",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    Whas this helpful?
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Whas this helpful?",
                  },
                  "ref": null,
                  "rendered": "Whas this helpful?",
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="btn positive "
                      onClick={[Function]}
                    >
                      <i
                        className="fas fa-thumbs-up"
                      />
                      <span>
                        YES
                      </span>
                    </button>,
                    <button
                      className="btn negative "
                      onClick={[Function]}
                    >
                      <i
                        className="fas fa-thumbs-down"
                      />
                      <span>
                        NO
                      </span>
                    </button>,
                  ],
                  "className": "buttons",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <i
                          className="fas fa-thumbs-up"
                        />,
                        <span>
                          YES
                        </span>,
                      ],
                      "className": "btn positive ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fas fa-thumbs-up",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "YES",
                        },
                        "ref": null,
                        "rendered": "YES",
                        "type": "span",
                      },
                    ],
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <i
                          className="fas fa-thumbs-down"
                        />,
                        <span>
                          NO
                        </span>,
                      ],
                      "className": "btn negative ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fas fa-thumbs-down",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "NO",
                        },
                        "ref": null,
                        "rendered": "NO",
                        "type": "span",
                      },
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="question"
            data-test="question-opener"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="far fa-plus-square fa-2x"
            />
            <h3>
              faq.ques1
            </h3>
          </div>,
          <div
            className="answer"
          >
            <div>
              <p>
                We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
              </p>
              <p>
                We will be adding more currencies very soon, stay tuned!
              </p>
              <p>
                Missing your favorite coin? Let us know here: 
                <a
                  href="mailto:support@nexchange.co.uk"
                >
                  support@nexchange.co.uk
                </a>
                .
              </p>
            </div>
            <div
              className="feedback"
            >
              <div>
                <span>
                  Whas this helpful?
                </span>
              </div>
              <div
                className="buttons"
              >
                <button
                  className="btn positive "
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-thumbs-up"
                  />
                  <span>
                    YES
                  </span>
                </button>
                <button
                  className="btn negative "
                  onClick={[Function]}
                >
                  <i
                    className="fas fa-thumbs-down"
                  />
                  <span>
                    NO
                  </span>
                </button>
              </div>
            </div>
          </div>,
          <hr />,
        ],
        "className": "question-answer container",
        "id": "faq.ques1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                aria-hidden="true"
                className="far fa-plus-square fa-2x"
              />,
              <h3>
                faq.ques1
              </h3>,
            ],
            "className": "question",
            "data-test": "question-opener",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "far fa-plus-square fa-2x",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "faq.ques1",
              },
              "ref": null,
              "rendered": "faq.ques1",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <p>
                  We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
                </p>
                <p>
                  We will be adding more currencies very soon, stay tuned!
                </p>
                <p>
                  Missing your favorite coin? Let us know here: 
                  <a
                    href="mailto:support@nexchange.co.uk"
                  >
                    support@nexchange.co.uk
                  </a>
                  .
                </p>
              </div>,
              <div
                className="feedback"
              >
                <div>
                  <span>
                    Whas this helpful?
                  </span>
                </div>
                <div
                  className="buttons"
                >
                  <button
                    className="btn positive "
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-thumbs-up"
                    />
                    <span>
                      YES
                    </span>
                  </button>
                  <button
                    className="btn negative "
                    onClick={[Function]}
                  >
                    <i
                      className="fas fa-thumbs-down"
                    />
                    <span>
                      NO
                    </span>
                  </button>
                </div>
              </div>,
            ],
            "className": "answer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.
                  </p>,
                  <p>
                    We will be adding more currencies very soon, stay tuned!
                  </p>,
                  <p>
                    Missing your favorite coin? Let us know here: 
                    <a
                      href="mailto:support@nexchange.co.uk"
                    >
                      support@nexchange.co.uk
                    </a>
                    .
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.",
                  },
                  "ref": null,
                  "rendered": "We allow you to exchange one cryptocurrency for another. To view our currently supported coins, please click on the coin selection widget in the top of fold of the page.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "We will be adding more currencies very soon, stay tuned!",
                  },
                  "ref": null,
                  "rendered": "We will be adding more currencies very soon, stay tuned!",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Missing your favorite coin? Let us know here: ",
                      <a
                        href="mailto:support@nexchange.co.uk"
                      >
                        support@nexchange.co.uk
                      </a>,
                      ".",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Missing your favorite coin? Let us know here: ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "support@nexchange.co.uk",
                        "href": "mailto:support@nexchange.co.uk",
                      },
                      "ref": null,
                      "rendered": "support@nexchange.co.uk",
                      "type": "a",
                    },
                    ".",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <span>
                      Whas this helpful?
                    </span>
                  </div>,
                  <div
                    className="buttons"
                  >
                    <button
                      className="btn positive "
                      onClick={[Function]}
                    >
                      <i
                        className="fas fa-thumbs-up"
                      />
                      <span>
                        YES
                      </span>
                    </button>
                    <button
                      className="btn negative "
                      onClick={[Function]}
                    >
                      <i
                        className="fas fa-thumbs-down"
                      />
                      <span>
                        NO
                      </span>
                    </button>
                  </div>,
                ],
                "className": "feedback",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                      Whas this helpful?
                    </span>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Whas this helpful?",
                    },
                    "ref": null,
                    "rendered": "Whas this helpful?",
                    "type": "span",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="btn positive "
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-thumbs-up"
                        />
                        <span>
                          YES
                        </span>
                      </button>,
                      <button
                        className="btn negative "
                        onClick={[Function]}
                      >
                        <i
                          className="fas fa-thumbs-down"
                        />
                        <span>
                          NO
                        </span>
                      </button>,
                    ],
                    "className": "buttons",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <i
                            className="fas fa-thumbs-up"
                          />,
                          <span>
                            YES
                          </span>,
                        ],
                        "className": "btn positive ",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fas fa-thumbs-up",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "YES",
                          },
                          "ref": null,
                          "rendered": "YES",
                          "type": "span",
                        },
                      ],
                      "type": "button",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <i
                            className="fas fa-thumbs-down"
                          />,
                          <span>
                            NO
                          </span>,
                        ],
                        "className": "btn negative ",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fas fa-thumbs-down",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "NO",
                          },
                          "ref": null,
                          "rendered": "NO",
                          "type": "span",
                        },
                      ],
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
    Symbol(enzyme.__providerValues__): Map {},
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
